src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java

```diff
@@ -75,14 +72,9 @@ public class SwerveSubsystem extends SubsystemBase {
 
   public Field2d field = new Field2d();
 
-  private boolean isInTapeMode = true;
-  private PIDController tapeDriveAssistController = new PIDController(-0.018, 0, -0.01);
-
-  private VisionIOApriltags apriltagVisionSubsystem = new VisionIOApriltags();
-  private VisionIOSimApriltags apriltagVisionSim = new VisionIOSimApriltags();
+  private VisionSubsystem visionSubsystem = new VisionSubsystem();
+  private VisionIOReal visionIOReal = new VisionIOReal();
```

Comment:
should only need the subsystem or the io, not both

Reply:
tbh you could probably get rid of the subsystem file entirely and just have an instance of `VisionIO` and an instance of `VisionIOInputs` in `SwerveSubsystem`